src/python/pants/goal/context.py

```diff
@@ -155,7 +155,37 @@ def __str__(self):
     ident = Target.identify(self.targets())
     return 'Context(id:{}, targets:{})'.format(ident, self.targets())
 
-  def set_resulting_graph_size_in_runtracker(self):
+  @contextmanager
+  def executing(self):
+    """A contextmanager that sets metrics in the context of a (v1) engine execution."""
+    self._set_target_root_count_in_runtracker()
+    yield
+    self._set_affected_target_count_in_runtracker()
+    self._set_affected_target_files_count_in_runtracker()
+    self._set_resulting_graph_size_in_runtracker()
+
+  def _set_target_root_count_in_runtracker(self):
+    """Sets the target root count in the run tracker's daemon stats object."""
+    # N.B. `self._target_roots` is always an expanded list of `Target` objects as
+    # provided by `GoalRunner`.
+    target_count = len(self._target_roots)
+    self.run_tracker.pantsd_stats.set_target_root_size(target_count)
+    return target_count
+
+  def _set_affected_target_count_in_runtracker(self):
+    """Sets the realized target count in the run tracker's daemon stats object."""
+    target_count = len(self.build_graph)
```

Comment:
We're going to be breaking this metric pretty soon, so I'm not sure we even want to begin collecting it.

Reply:
hmm, but only for the python related tasks afaict. my thinking is that it's still a fairly relevant and interesting grouping metric and one that might make sense to be able to compare/contrast file count to. ftr: I'm still not completely convinced that "files" is that much better than "targets" since they can both be deceptive as to substantive size on the surface. so was thinking both can't hurt, initially - and it's cheap to collect/report on.

wdyt?

Reply:
> we essentially pre-deprecate this metric and bake in the assumption that only v1 tasks report it

yeah, that's the idea. this metric is exactly as "deprecated" as `Target`/`BuildGraph` is. once we erode `Target` et al in favor of pushing more into the v2 engine, I fully expect this metric to also erode/evolve alongside it - but think it might be useful to experiment with in the interim.

Reply:
Ok. Fine with landing it.

> hmm, but only for the python related tasks afaict.

The first task it will be going away for is `list` in #5639